xbmc/utils/StringUtils.cpp

```diff
@@ -1894,7 +2188,4455 @@ const std::locale& StringUtils::GetOriginalLocale() noexcept
+    CaseMap(U'\x10c81', U'\x00000', U'\x10cc1', U'\x00000'),
+    CaseMap(U'\x10c82', U'\x00000', U'\x10cc2', U'\x00000'),
+    CaseMap(U'\x10c83', U'\x00000', U'\x10cc3', U'\x00000'),
+    CaseMap(U'\x10c84', U'\x00000', U'\x10cc4', U'\x00000'),
+    CaseMap(U'\x10c85', U'\x00000', U'\x10cc5', U'\x00000'),
+    CaseMap(U'\x10c86', U'\x00000', U'\x10cc6', U'\x00000'),
+    CaseMap(U'\x10c87', U'\x00000', U'\x10cc7', U'\x00000'),
+    CaseMap(U'\x10c88', U'\x00000', U'\x10cc8', U'\x00000'),
+    CaseMap(U'\x10c89', U'\x00000', U'\x10cc9', U'\x00000'),
+    CaseMap(U'\x10c8a', U'\x00000', U'\x10cca', U'\x00000'),
+    CaseMap(U'\x10c8b', U'\x00000', U'\x10ccb', U'\x00000'),
+    CaseMap(U'\x10c8c', U'\x00000', U'\x10ccc', U'\x00000'),
+    CaseMap(U'\x10c8d', U'\x00000', U'\x10ccd', U'\x00000'),
+    CaseMap(U'\x10c8e', U'\x00000', U'\x10cce', U'\x00000'),
+    CaseMap(U'\x10c8f', U'\x00000', U'\x10ccf', U'\x00000'),
+    CaseMap(U'\x10c90', U'\x00000', U'\x10cd0', U'\x00000'),
+    CaseMap(U'\x10c91', U'\x00000', U'\x10cd1', U'\x00000'),
+    CaseMap(U'\x10c92', U'\x00000', U'\x10cd2', U'\x00000'),
+    CaseMap(U'\x10c93', U'\x00000', U'\x10cd3', U'\x00000'),
+    CaseMap(U'\x10c94', U'\x00000', U'\x10cd4', U'\x00000'),
+    CaseMap(U'\x10c95', U'\x00000', U'\x10cd5', U'\x00000'),
+    CaseMap(U'\x10c96', U'\x00000', U'\x10cd6', U'\x00000'),
+    CaseMap(U'\x10c97', U'\x00000', U'\x10cd7', U'\x00000'),
+    CaseMap(U'\x10c98', U'\x00000', U'\x10cd8', U'\x00000'),
+    CaseMap(U'\x10c99', U'\x00000', U'\x10cd9', U'\x00000'),
+    CaseMap(U'\x10c9a', U'\x00000', U'\x10cda', U'\x00000'),
+    CaseMap(U'\x10c9b', U'\x00000', U'\x10cdb', U'\x00000'),
+    CaseMap(U'\x10c9c', U'\x00000', U'\x10cdc', U'\x00000'),
+    CaseMap(U'\x10c9d', U'\x00000', U'\x10cdd', U'\x00000'),
+    CaseMap(U'\x10c9e', U'\x00000', U'\x10cde', U'\x00000'),
+    CaseMap(U'\x10c9f', U'\x00000', U'\x10cdf', U'\x00000'),
+    CaseMap(U'\x10ca0', U'\x00000', U'\x10ce0', U'\x00000'),
+    CaseMap(U'\x10ca1', U'\x00000', U'\x10ce1', U'\x00000'),
+    CaseMap(U'\x10ca2', U'\x00000', U'\x10ce2', U'\x00000'),
+    CaseMap(U'\x10ca3', U'\x00000', U'\x10ce3', U'\x00000'),
+    CaseMap(U'\x10ca4', U'\x00000', U'\x10ce4', U'\x00000'),
+    CaseMap(U'\x10ca5', U'\x00000', U'\x10ce5', U'\x00000'),
+    CaseMap(U'\x10ca6', U'\x00000', U'\x10ce6', U'\x00000'),
+    CaseMap(U'\x10ca7', U'\x00000', U'\x10ce7', U'\x00000'),
+    CaseMap(U'\x10ca8', U'\x00000', U'\x10ce8', U'\x00000'),
+    CaseMap(U'\x10ca9', U'\x00000', U'\x10ce9', U'\x00000'),
+    CaseMap(U'\x10caa', U'\x00000', U'\x10cea', U'\x00000'),
+    CaseMap(U'\x10cab', U'\x00000', U'\x10ceb', U'\x00000'),
+    CaseMap(U'\x10cac', U'\x00000', U'\x10cec', U'\x00000'),
+    CaseMap(U'\x10cad', U'\x00000', U'\x10ced', U'\x00000'),
+    CaseMap(U'\x10cae', U'\x00000', U'\x10cee', U'\x00000'),
+    CaseMap(U'\x10caf', U'\x00000', U'\x10cef', U'\x00000'),
+    CaseMap(U'\x10cb0', U'\
```

Comment:
this will make a copy of the whole table... please use a pointer or similar

Reply:
Recoded all of the tables to use constexpr.